src/Workspaces/Core/Portable/SyntaxDifferencing/SyntaxMatch.cs

```diff
@@ -6,9 +6,9 @@
 using System.Diagnostics;
 using System.Linq;
 
-namespace Microsoft.CodeAnalysis.Differencing
+namespace Microsoft.CodeAnalysis.SyntaxDifferencing
 {
-    public sealed partial class Match<TNode>
+    internal sealed partial class SyntaxMatch
```

Comment:
Currently, this type would become public.  However, i only want to do that if a large subset of functionality on this type is needed.  If only a subset of functionality is needed, i would rather just expose that.